### PR TITLE
mwlwifi: update to version 10.4.10-20240419

### PR DESCRIPTION
**Pull request by stklcode**
This update contains a minor fix to resolve "detected write beyond size of field" warning during compilation:
* "replace [0] with []" ([1d0d08c](https://github.com/kaloz/mwlwifi/commit/af661d9260d6d732cde384fbe2281d7653c68496))

All patches still apply.

References:
* https://github.com/openwrt/openwrt/issues/15108

----

Reported by @anomeome, upstream fix by @neheb 

Full upstream diff: https://github.com/kaloz/mwlwifi/compare/ebf31674...a737d348

Build and run tested on WRT1900ACS (mvebu/cortexa9)

### CODE REVIEW

File: package/kernel/mwlwifi/Makefile [hunk 1]
```diff
@@ -15,9 +15,9 @@ PKG_LICENSE_FILES:=
 
 PKG_SOURCE_URL:=https://github.com/kaloz/mwlwifi
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_DATE:=2023-11-29
-PKG_SOURCE_VERSION:=ebf3167445f108346dcff9a31a708534c0bd7cc5
-PKG_MIRROR_HASH:=6dffc08208305366b9c25133b74693d5c2e544c1076b7b4ed846628c66be56c1
+PKG_SOURCE_DATE:=2024-04-19
```
Comment (neheb) [inline package/kernel/mwlwifi/Makefile:18]: IIRC apk requires replacing - with .

Comment (stklcode) [reply package/kernel/mwlwifi/Makefile:18]: This is done while generating `PKG_VERSION` from `PKG_SOURCE_DATE`.

https://github.com/openwrt/openwrt/blob/ea609fe486a99af6540dcc7a18f0827481d0c4b0/include/download.mk#L13

I’m fine with replacing it directly, but is seems most packages use `YYYY-MM-DD` format for source date.

Comment (neheb) [reply package/kernel/mwlwifi/Makefile:18]: Ah OK nvm.

**Review by neheb — APPROVED** [review]

**Review by dynasticorpheus — APPROVED** [review]

**Comment by robimarko** [comment]
@stklcode Well, this fixes the pointed compile error, but when building for mvebu/cortex-a53 I get another error.
```
  CC [M]  /home/robimarko/Building/mwlwifi-openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mvebu_cortexa53/mwlwifi-2024.04.19~a737d348/debugfs.o
/home/robimarko/Building/mwlwifi-openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mvebu_cortexa53/mwlwifi-2024.04.19~a737d348/debugfs.c: In function 'mwl_debugfs_info_read':
/home/robimarko/Building/mwlwifi-openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mvebu_cortexa53/mwlwifi-2024.04.19~a737d348/debugfs.c:367:99: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
  367 |                                 "wcb_base0   : %x => %8x|%8p|%4d|%d\n", get_hw_spec->wcb_base0, *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),skb_queue_len(&pcie_priv->txq[0]),pcie_priv->fw_desc_cnt[0]);
      |                                                                                                   ^
/home/robimarko/Building/mwlwifi-openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mvebu_cortexa53/mwlwifi-2024.04.19~a737d348/debugfs.c:367:162: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
  367 |                                 "wcb_base0   : %x => %8x|%8p|%4d|%d\n", get_hw_spec->wcb_base0, *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),skb_queue_len(&pcie_priv->txq[0]),pcie_priv->fw_desc_cnt[0]);
      |                                                                                                                                                                  ^
/home/robimarko/Building/mwlwifi-openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mvebu_cortexa53/mwlwifi-2024.04.19~a737d348/debugfs.c:367:152: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
  367 |                                 "wcb_base0   : %x => %8x|%8p|%4d|%d\n", get_hw_spec->wcb_base0, *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),skb_queue_len(&pcie_priv->txq[0]),pcie_priv->fw_desc_cnt[0]);
      |                                                                                                                                                        ^
/home/robimarko/Building/mwlwifi-openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mvebu_cortexa53/mwlwifi-2024.04.19~a737d348/debugfs.c:370:105: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
  370 |                                 "wcb_base[%2d]: %x => %8x|%8p|%4d|%d\n", i, get_hw_spec->wcb_base[i], *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base[i])),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base[i])),skb_queue_len(&pcie_priv->txq[i + 1]),pcie_priv->fw_desc_cnt[i + 1]);
      |                                                                                                         ^
/home/robimarko/Building/mwlwifi-openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mvebu_cortexa53/mwlwifi-2024.04.19~a737d348/debugfs.c:370:170: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
  370 |                                 "wcb_base[%2d]: %x => %8x|%8p|%4d|%d\n", i, get_hw_spec->wcb_base[i], *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base[i])),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base[i])),skb_queue_len(&pcie_priv->txq[i + 1]),pcie_priv->fw_desc_cnt[i + 1]);
      |                                                                                                                                                                          ^
/home/robimarko/Building/mwlwifi-openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mvebu_cortexa53/mwlwifi-2024.04.19~a737d348/debugfs.c:370:160: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
  370 |                                 "wcb_base[%2d]: %x => %8x|%8p|%4d|%d\n", i, get_hw_spec->wcb_base[i], *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base[i])),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base[i])),skb_queue_len(&pcie_priv->txq[i + 1]),pcie_priv->fw_desc_cnt[i + 1]);
      |                                                                                                                                                                ^
cc1: all warnings being treated as errors
make[5]: *** [scripts/Makefile.build:243: /home/robimarko/Building/mwlwifi-openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mvebu_cortexa53/mwlwifi-2024.04.19~a737d348/debugfs.o] Error 1
```

**Comment by robimarko** [comment]
Thanks! Rebased on top of main and merged!

**Comment by neheb** [comment]
```
--- a/debugfs.c
+++ b/debugfs.c
@@ -364,10 +364,10 @@ static ssize_t mwl_debugfs_info_read(struct file *file, char __user *ubuf,
                         "-----------------------=>  address| address|qlen|fw_desc_cnt\n");
                spin_lock_irqsave(&pcie_priv->tx_desc_lock, flags);
                len += scnprintf(p + len, size - len,
-                               "wcb_base0   : %x => %8x|%8p|%4d|%d\n", get_hw_spec->wcb_base0, *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),skb_queue_len(&pcie_priv->txq[0]),pcie_priv->fw_desc_cnt[0]);
+                               "wcb_base0   : %x => %8x|%8p|%4d|%d\n", get_hw_spec->wcb_base0, le32_to_cpu(get_hw_spec->wcb_base0),&get_hw_spec->wcb_base0,skb_queue_len(&pcie_priv->txq[0]),pcie_priv->fw_desc_cnt[0]);
                for(i = 0; i < SYSADPT_TOTAL_TX_QUEUES - 1; i++)
                        len += scnprintf(p + len, size - len,
-                               "wcb_base[%2d]: %x => %8x|%8p|%4d|%d\n", i, get_hw_spec->wcb_base[i], *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base[i])),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base[i])),skb_queue_len(&pcie_priv->txq[i + 1]),pcie_priv->fw_desc_cnt[i + 1]);
+                               "wcb_base[%2d]: %x => %8x|%8p|%4d|%d\n", i, get_hw_spec->wcb_base[i], get_hw_spec->wcb_base[i],&get_hw_spec->wcb_base[i],skb_queue_len(&pcie_priv->txq[i + 1]),pcie_priv->fw_desc_cnt[i + 1]);
                spin_unlock_irqrestore(&pcie_priv->tx_desc_lock, flags);
        }
 

```

Change is 100% wrong but it does compile.

**Pull request merged**